carrot-log4j-aws-sns/src/main/java/com/carrotgarden/log4j/aws/sns/Appender.java

```diff
@@ -216,14 +225,14 @@ protected boolean ensureCredentials() {
 
             if (url != null) {
                 try {
-                    amazonCredentials = new PropertiesCredentials(url.openStream());
+                    amazonCredentials = new StaticCredentialsProvider(new PropertiesCredentials(url.openStream()));
                     return true;
                 } catch (Exception e) {
                     // fail out
                 }
             } else {
                 try {
-                    amazonCredentials = new PropertiesCredentials(new File(getCredentials()));
+                    amazonCredentials = new StaticCredentialsProvider(new PropertiesCredentials(new File(getCredentials())));
```

Comment:
I'm hinky about using anything from com.amazonaws.internal, they don't even show up in their JavaDoc. Could we use PropertiesFileCredentialsProvider(getCredentials()) instead?